File: client/java/src/avt/Command.java
package avt;

import javax.microedition.lcdui.Graphics;
import main.Canvas;

public class Command {
   public String caption;
   public IAction action;
   public byte indexMenu;
   public AvMain pointer;
   public short subIndex = -1;

   public Command(String var1, IAction var2) {
      this.caption = var1;
      this.action = var2;
   }

   public Command(String var1, int var2) {
      this.caption = var1;
      this.indexMenu = (byte)var2;
   }

   public Command(String var1, int var2, AvMain var3) {
      this.caption = var1;
      this.indexMenu = (byte)var2;
      this.pointer = var3;
   }

   public Command(String var1, int var2, int var3) {
      this.caption = var1;
      this.indexMenu = (byte)var2;
      this.subIndex = (short)((byte)var3);
   }

   public final void perform() {
      if (this.action != null) {
         this.action.perform();
      } else if (this.pointer != null) {
         this.pointer.commandActionPointer(this.indexMenu);
      } else if (ChatTextField.isShow) {
         ChatTextField.gI().commandTab(this.indexMenu, this.subIndex);
      } else {
         Canvas.currentMyScreen.commandTab(this.indexMenu, this.subIndex);
      }

   }

   public void update() {
   }

   public void paint(Graphics var1, int var2, int var3) {
      Canvas.borderFont.drawString(var1, this.caption, var2, var3, 2);
   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/Command.cs
public class Command
{
	public string caption;

	public IAction action;

	public IKbAction ipaction;

	public sbyte indexMenu;

	public sbyte subIndex;

	public sbyte indexImage;

	public int x = -1;

	public int y = -1;

	public AvMain pointer;

	public Command()
	{
	}

	public Command(string caption, IAction action)
	{
		this.caption = caption;
		this.action = action;
	}

	public Command(string caption, IKbAction action)
	{
		this.caption = caption;
		ipaction = action;
	}

	public Command(string caption, int type)
	{
		this.caption = caption;
		indexMenu = (sbyte)type;
	}

	public Command(string caption, int type, AvMain pointer)
	{
		this.caption = caption;
		indexMenu = (sbyte)type;
		this.pointer = pointer;
	}

	public Command(string caption, int type, int sub)
	{
		this.caption = caption;
		indexMenu = (sbyte)type;
		subIndex = (sbyte)sub;
	}

	public Command(string caption, int type, int subIndex, AvMain pointer)
	{
		this.caption = caption;
		indexMenu = (sbyte)type;
		this.pointer = pointer;
		this.subIndex = (sbyte)subIndex;
	}

	public Command(string caption, int type, int x, int y)
	{
		this.caption = caption;
		indexMenu = (sbyte)type;
		this.x = x;
		this.y = y;
	}

	public Command(string caption, IAction action, int x, int y)
	{
		this.caption = caption;
		this.action = action;
		this.x = x;
		this.y = y;
	}

	public void perform()
	{
		Out.println("command gettext: " + Canvas.inputDlg.tfInput.getText());
		if (action != null)
		{
			action.perform();
		}
		else if (ipaction != null)
		{
			ipaction.perform(Canvas.inputDlg.tfInput.getText());
		}
		else if (pointer != null)
		{
			pointer.commandActionPointer(indexMenu, subIndex);
		}
		else if (Canvas.currentDialog != null)
		{
			Canvas.currentDialog.commandTab(indexMenu);
		}
		else
		{
			Canvas.currentMyScreen.commandTab(indexMenu);
		}
	}

	public virtual void update()
	{
	}

	public virtual void paint(MyGraphics g, int x, int y)
	{
	}
}
